src/server/lib/http-request.js

```diff
@@ -0,0 +1,123 @@
+import originalFetch from "node-fetch";
+import AbortController from "node-abort-controller";
+import { log } from "../../lib";
+import { v4 as uuid } from "uuid";
+
+const requestWithRetry = async (url, props) => {
+  const originalProps = props || {};
+  const remainingProps = {
+    ...(originalProps || {})
+  };
+  delete remainingProps.validateStatus;
+  delete remainingProps.retries;
+  delete remainingProps.timeout;
+
+  const requestId = uuid();
+  const propsRetries = originalProps.retries;
+  const retries =
+    propsRetries !== null && propsRetries !== undefined ? propsRetries : 2;
+  const timeout = originalProps.timeout || 2000;
+
+  const retryDelay = () => {
+    const baseDelay = 50;
+    const randomDelay = Math.floor(Math.random() * (baseDelay / 2));
```

Comment:
Why incorporate randomness here?

Reply:
We found it to be a best practice at my day job.